apsuite/lattice_errors_new.py

```diff
@@ -0,0 +1,1539 @@
+#!/usr/bin/env python-sirius
+
+import numpy as _np
+import matplotlib.pyplot as _plt
+import time as _time
+import copy as _copy
+
+import pyaccel as _pyaccel
+import pymodels as _pymodels
+from apsuite.orbcorr import OrbitCorr, CorrParams
+from apsuite.optics_analysis import TuneCorr, OpticsCorr, CouplingCorr
+from apsuite.commisslib.measure_bba import BBAParams
+from mathphys.functions import save_pickle, load_pickle
+
+
+class ConfigErrors:
+    """Class for configure general types of errors."""
+
+    def __init__(self):
+        """Errors attributes"""
+        self._fam_names = []
+        self._sigma_x = 0
+        self._sigma_y = 0
+        self._sigma_roll = 0
+        self._sigma_pitch = 0
+        self._sigma_yaw = 0
+        self._sigmas = dict()
+        self._um = 1e-6
+        self._mrad = 1e-3
+        self._percent = 1e-2
+
+    @property
+    def fam_names(self):
+        """Families which the errors are going to be applied.
+
+        Returns:
+            list of string: List with the names of the families
+        """
+        return self._fam_names
+
+    @fam_names.setter
+    def fam_names(self, value):
+        self._fam_names = value
+
+    @property
+    def sigma_x(self):
+        """Standard deviation of the x position errors.
+
+        Returns:
+            float: [um]
+        """
+        return self._sigma_x
+
+    @sigma_x.setter
+    def sigma_x(self, value):
+        self._sigma_x = value * self._um
+
+    @property
+    def sigma_y(self):
+        """Standard deviation of the y position errors.
+
+        Returns:
+            float: [um]
+        """
+        return self._sigma_y
+
+    @sigma_y.setter
+    def sigma_y(self, value):
+        self._sigma_y = value * self._um
+
+    @property
+    def sigma_roll(self):
+        """Standard deviation of the roll errors.
+
+        Returns:
+            float: [mrad]
+        """
+        return self._sigma_roll
+
+    @sigma_roll.setter
+    def sigma_roll(self, value):
+        self._sigma_roll = value * self._mrad
+
+    @property
+    def sigma_pitch(self):
+        """Standard deviation of the pitch errors.
+
+        Returns:
+            float: [mrad]
+        """
+        return self._sigma_pitch
+
+    @sigma_pitch.setter
+    def sigma_pitch(self, value):
+        self._sigma_pitch = value * self._mrad
+
+    @property
+    def sigma_yaw(self):
+        """Standard deviation of the yaw errors.
+
+        Returns:
+            float: [mrad]
+        """
+        return self._sigma_yaw
+
+    @sigma_yaw.setter
+    def sigma_yaw(self, value):
+        self._sigma_yaw = value * self._mrad
+
+    @property
+    def sigmas(self):
+        """Dictionary with names and values of the configured errors.
+
+        Returns:
+            dictionary: Keys are the types of the errors, values are the std.
+        """
+        return self._sigmas
+
+    @sigmas.setter
+    def sigmas(self, value):
```

Comment:
I agree that it probably is not the best name, do you have a suggestion? We must be careful with the names because we have this dictionary that just contains the type of error and its standard deviation, and we also have the fam_errors dictionary, which contains the errors themselves grouped by family and type of error.

Reply:
What about sigmas_dict?